byway/src/main/java/com/google/sps/servlets/DestinationsServlet.java

```diff
@@ -30,52 +31,53 @@
 
   private final Gson gson = new Gson();
   private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-  private Key userKey;
-
-  @Override
-  public void init() {
-    Entity userEntity = new Entity("UserInputs");
-    userEntity.setProperty("start", "");
-    userEntity.setProperty("destinations", new ArrayList<String>());
-    datastore.put(userEntity);
-    userKey= userEntity.getKey();
-  }
-
+  private final UserService userService = UserServiceFactory.getUserService();
+  private Key tripKey;
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    tripKey = KeyFactory.stringToKey(request.getParameter("tripKey"));
```

Comment:
What if the trip key parameter is empty? Where is the key being generated? It seems it would be best to handle creating new trip entities (and therefore tripKeys) inside this class. 

You could have this method work with or without the query parameter for tripKey. If one is specified, it loads the existing key. If one is not provided, it could create a new Trip entity, add it to the current user's info, and return the new trip. Subsequent requests would provide the query parameter to reference the new object.

Reply:
i have the trip being creating in EntityServlet.java now